Tasks/DotNetCoreInstallerV1/versionFetcher.ts

```diff
@@ -0,0 +1,273 @@
+                })
+                .then((body: string) => {
+                    channelReleases = JSON.parse(body).releases;
+                    if (versionParts.minorVersion == "x" || versionParts.patchVersion == "x") {
+                        let dotNetSdkVersionTelemetry = `{"version":"${version}"}`;
+                        console.log("##vso[telemetry.publish area=TaskDeploymentMethod;feature=DotNetCoreInstallerV1]" + dotNetSdkVersionTelemetry);
+
+                        let latestVersion = "0.0.0";
+                        channelReleases.forEach(release => {
+                            if (release[packageType] && utils.versionCompareFunction(release[packageType].version, latestVersion) > 0 && (includePreviewVersions || !release[packageType].version.includes('preview'))) {
+                                latestVersion = release[packageType].version;
+                            }
+                        });
+
+                        if (latestVersion == "0.0.0") {
+                            throw tl.loc("MatchingVersionNotFound", version);
+                        }
+
+                        console.log(tl.loc("MatchingVersionForUserInputVersion", version, latestVersion));
+                        version = latestVersion;
+                    }
+
+                    var versionRelease = channelReleases.find(release => {
+                        if (release[packageType] && release[packageType].version == version) {
+                            return true;
+                        }
+
+                        return false;
+                    });
+
+                    return versionRelease ? versionRelease[packageType] : null;
+                });
+        }
+        else {
+            throw tl.loc("UrlForReleaseChannelNotFound");
+        }
+    }
+
+    private async getVersionFromOtherChannels(version: string, packageType: string, includePreviewVersions: boolean): Promise<VersionInfo> {
+        let fallbackChannels = this.getChannelsForMajorVersion(version);
+        if (!fallbackChannels && fallbackChannels.length < 1) {
+            throw tl.loc("NoSuitableChannelWereFound", version);
+        }
+
+        var versionInfo: VersionInfo = null;
+        for (var i = 0; i < fallbackChannels.length; i++) {
+            console.log("LookingForVersionInChannel", (fallbackChannels[i])["channel-version"]);
+            versionInfo = await this.getVersionFromChannel(fallbackChannels[i], version, packageType, includePreviewVersions);
+
+            if (versionInfo) {
+                break;
+            }
+        }
+
+        return versionInfo;
+    }
+
+    private getChannelsForMajorVersion(version: string): any {
+        var versionParts = utils.getVersionParts(version);
+        let adjacentChannels = [];
+        this.releasesIndex["releases-index"].forEach(channel => {
+            if (channel["channel-version"].startsWith(`${versionParts.majorVersion}`)) {
+                adjacentChannels.push(channel);
```

Comment:
When is this valid?

Reply:
Some versions such as 2.1.100 - 104 are present in 2.0 channel and not in 2.1 channel.
This flow is only exercised when user types an exact version and not in case of finding latest versions.

Hence when the user enters `version = 2.1.102`, we will not be able to find the version in 2.1 channel. We will then fallback to finding this version in all channels which are like 2.x (2.0, 2.1, 2.2). The version will then be found in 2.0